openedx/core/djangoapps/public_api/courses/tests/utils.py

```diff
@@ -0,0 +1,101 @@
+# pylint: disable=E1101
+'''
+Utilities for contentstore tests
+'''
+
+from datetime import timedelta
+
+from django.utils import timezone
+from rest_framework.test import APIClient
+
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from django.conf import settings
+
+from provider.oauth2.models import AccessToken, Client as OAuth2Client
+from provider import constants
+
+TEST_DATA_DIR = settings.COMMON_TEST_DATA_ROOT
+
+
+def create_oauth2_client(user):
+    """
+    Create an OAuth2 client associated with the given user and generate an
+    access token for said client.
+
+    :param user:
+    :return: a Client (provider.oauth2) and an AccessToken
+    """
+    # Register an OAuth2 Client
+    client = OAuth2Client(
+        user=user,
+        name=user.username,
+        url="http://127.0.0.1/",
+        redirect_uri="http://127.0.0.1/",
+        client_type=constants.CONFIDENTIAL
+    )
+    client.save()
+
+    # Generate an access token for the client
+    access_token = AccessToken(
+        user=user,
+        client=client,
+
+        # Set the access token to expire one day from now
+        expires=timezone.now() + timedelta(1, 0),
+        scope=constants.READ_WRITE
+    )
+    access_token.save()
+
+    return client, access_token
+
+
+def use_access_token(client, access_token):
+    """
+    Make an APIClient pass an access token for all requests
+
+    :param client: an APIClient
+    :param access_token: an AccessToken
+    """
+    client.credentials(
+        HTTP_AUTHORIZATION="Bearer {}".format(access_token.token)
+    )
```

Comment:
Instead of having this function rely on side effects, can you at least have it return the `client` object, and have functions that call it use the new `client` object that this returns?